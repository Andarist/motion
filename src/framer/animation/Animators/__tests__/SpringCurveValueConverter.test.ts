import { SpringCurveValueConverter } from "../SpringCurveValueConverter"

describe("computeDampingRatio", () => {
    it("should return correct value", () => {
        expect(SpringCurveValueConverter.computeDampingRatio(10, 12, 2)).toMatchSnapshot()
        expect(SpringCurveValueConverter.computeDampingRatio(10, 12)).toMatchSnapshot()
    })
})
describe("computeDuration", () => {
    it("should return correct value", () => {
        expect(SpringCurveValueConverter.computeDuration(1, 1, 0, 1)).toMatchSnapshot()
        expect(SpringCurveValueConverter.computeDuration(1, 1)).toMatchSnapshot()
    })
    it("should return null", () => {
        expect(SpringCurveValueConverter.computeDuration(10, 10, 0, 1)).toMatchSnapshot()
        expect(SpringCurveValueConverter.computeDuration(10, 10)).toMatchSnapshot()
    })
})
describe("computeDerivedCurveOptions", () => {
    it("should return correct value", () => {
        expect(SpringCurveValueConverter.computeDerivedCurveOptions(1, 1, 0, 1)).toMatchSnapshot()
        expect(SpringCurveValueConverter.computeDerivedCurveOptions(1, 1)).toMatchSnapshot()
        expect(SpringCurveValueConverter.computeDerivedCurveOptions(0.1, 1)).toMatchSnapshot()
    })
})
