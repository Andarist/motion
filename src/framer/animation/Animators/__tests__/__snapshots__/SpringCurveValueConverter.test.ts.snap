// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`computeDampingRatio should return correct value 1`] = `1.3416407864998738`;

exports[`computeDampingRatio should return correct value 2`] = `1.8973665961010275`;

exports[`computeDerivedCurveOptions should return correct value 1`] = `
Object {
  "friction": 18.466826952903173,
  "tension": 85.25592442711776,
  "velocity": 0,
}
`;

exports[`computeDerivedCurveOptions should return correct value 2`] = `
Object {
  "friction": 18.466826952903173,
  "tension": 85.25592442711776,
  "velocity": 0,
}
`;

exports[`computeDerivedCurveOptions should return correct value 3`] = `
Object {
  "friction": 9.220390707829685,
  "tension": 2125.3901201257995,
  "velocity": 0,
}
`;

exports[`computeDuration should return correct value 1`] = `12.716898269296165`;

exports[`computeDuration should return correct value 2`] = `12.716898269296165`;

exports[`computeDuration should return null 1`] = `null`;

exports[`computeDuration should return null 2`] = `null`;
